Multisig/UI/Safe Management/Create Safe/Setup Safe/CreateSafeViewController.swift

```diff
@@ -572,37 +599,142 @@ class CreateSafeViewController: UIViewController, UITableViewDelegate, UITableVi
-        tableCell.tableView.registerCell(SecondaryDetailDisclosureCell.self)
+        let accountState: MiniAccountInfoUIModel = MiniAccountInfoUIModel(
+            address: uiModel.deployerAccountInfoModel?.address ?? Address(0),
+            label: uiModel.deployerAccountInfoModel?.label,
+            balance: uiModel.deployerAccountInfoModel?.balance
+            //TODO: add remaining fields
+        )
+        executionOptions.accountState = .filled(accountState)
 
-        let accountCell = deployerAccountCell(tableView: tableCell.tableView)
-        let estimatedFeeCell = estimateFeeCell(tableView: tableCell.tableView)
+        //        let relayerModel: RelayerInfoUIModel = RelayerInfoUIModel(remainingRelays: 2, limit: 5)
+        // TODO: Get values from server
+        //        executionOptions.relayerState = .filled(relayerModel)
 
-        tableCell.setCells([accountCell, estimatedFeeCell])
+        let cell = executionOptionsCellBuilder.buildExecutionOptions(executionOptions)[0]
 
-        // handle cell taps
-        let (executeWithIndex, feeIndex) = (0, 1)
-        tableCell.onCellTap = { [weak self] index in
-            guard let self = self else { return }
-            switch index {
-            case executeWithIndex:
-                self.selectDeploymentKey()
-            case feeIndex:
-                self.editParameters()
-            default:
-                assertionFailure("Tapped cell at index out of bounds: \(index)")
+        return cell
+    }
+
+    @IBAction func didTapPaymentMethod(_ sender: Any) {
+        // open payment method selection
+        let choosePaymentVC = ChoosePaymentViewController()
+        choosePaymentVC.relaysRemaining = relaysRemaining
+        choosePaymentVC.relaysLimit = relaysLimit
+
+        choosePaymentVC.chooseRelay = { [unowned self] in
+            LogService.shared.debug("User selected Relay")
+            executionOptionsCellBuilder.userSelectedSigner = false
+
+            tableView.reloadData()
+        }
+
+        choosePaymentVC.chooseSigner = { [unowned self] in
+            LogService.shared.debug("User selected Signer")
+            executionOptionsCellBuilder.userSelectedSigner = true
+            if self.uiModel.executionKeys().isEmpty {
+                let addOwnerVC = AddOwnerFirstViewController()
+                addOwnerVC.trackingEvent = .createSafeAddDeploymentKey
+                addOwnerVC.onSuccess = { [weak self] in
+                    guard let self = self else { return }
+                    self.navigationController?.popToViewController(self, animated: true)
+                    self.uiModel.selectedKey = self.uiModel.executionKeys().first
+                    self.uiModel.didEdit()
+                    Tracker.trackEvent(.createSafeDeploymentKeyAdded)
+                }
+                addOwnerVC.showsCloseButton = false
+                show(addOwnerVC, sender: self)
+                return
             }
+            // refresh ui
```

Comment:
I think this part is not complete ? reload data is not needed in case of updateUI, and we need to estimate the transaction again after changing payment method